appcues/src/main/java/com/appcues/AppcuesFirebaseMessagingService.kt

```diff
@@ -0,0 +1,100 @@
+package com.appcues
+
+import android.Manifest.permission
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import java.io.IOException
+import java.net.URL
+
+public class AppcuesFirebaseMessagingService : FirebaseMessagingService() {
+
+    public companion object {
+
+        // Suppressing MissingPermission for POST_NOTIFICATION since this should be done on customer's end.
+        @SuppressLint("MissingPermission")
+        @JvmStatic
+        public fun handleMessage(context: Context, message: RemoteMessage): Boolean {
```

Comment:
static method to allow for custom service to call similar and check if we can handle that message.

```kotlin
public void onMessageReceived(RemoteMessage remoteMessage) {
    super.onMessageReceived(remoteMessage);
    if (AppcuesFirebaseMessagingService.handleMessage(this, remoteMessage)) {
      // appcues handled message
    } else {
      // message from other messaging provider
    }
  }
```